app/src/main/java/com/example/cinemapp/ui/main/ProfileFragment.kt

```diff
@@ -0,0 +1,71 @@
+package com.example.cinemapp.ui.main
+
+import android.content.Intent
+import androidx.fragment.app.viewModels
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.lifecycle.lifecycleScope
+import com.example.cinemapp.R
+import com.example.cinemapp.data.UserPreferences
+import com.example.cinemapp.databinding.FragmentProfileBinding
+import com.example.cinemapp.databinding.FragmentSearchBinding
+import com.example.cinemapp.ui.authentication.AuthenticationActivity
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.koin.android.ext.android.inject
+
+class ProfileFragment : Fragment() {
+
+    private val viewModel: ProfileViewModel by viewModels()
+    private var _binding: FragmentProfileBinding? = null
+    private val binding get() = _binding!!
+    private val userPrefs: UserPreferences by inject()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentProfileBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onStart() {
+        super.onStart()
+        setupViews()
+    }
+
+    private fun setupViews() {
+        with(binding) {
+            btnSignOut.visibility = View.INVISIBLE
+            lifecycleScope.launch(Dispatchers.Main) {
+                userPrefs.getSessionId().collect { sessionId ->
+                    tvUsername.text = sessionId ?: "Not signed in"
+
+                    sessionId?.let {
+                        btnSignOut.visibility = View.VISIBLE
+                        btnSignOut.setOnClickListener {
```

Comment:
A lot of nesting here 😅 move `btnSignOut.setOnClickListener`should not depend on getting sessionId, but upon clicking on Sign Out button, ViewModel should handle logic of signing out, and expose something like `val signOutEvent: MutableSharedFlow<Unit>` which emits an event when user gets signed out

Reply:
I moved the listener, but for the rest, I planned to do that in the next ticket, for the profile fragment. This was just a temporary solution so that logging out would be possible at all.